packages/mad_pay_ios/ios/Classes/PaymentHelpers.swift

```diff
@@ -97,19 +97,31 @@ class PaymentNetworkHelper {
         paymentNetwork.compactMap({ decodePaymentNetwork($0) })
     }
     
-    static func getMerchantCapabilities(_ merchantCapabilities: Apple_MerchantCapabilities) -> PKMerchantCapability {
-        switch (merchantCapabilities) {
-        case Apple_MerchantCapabilities.threeds:
-            return PKMerchantCapability.capability3DS
-        case Apple_MerchantCapabilities.credit:
-            return PKMerchantCapability.capabilityCredit
-        case Apple_MerchantCapabilities.debit:
-            return PKMerchantCapability.capabilityDebit
-        case Apple_MerchantCapabilities.emv:
-            return PKMerchantCapability.capabilityEMV
-        default:
-            return PKMerchantCapability.capability3DS
+    static func getMerchantCapabilities(_ merchantCapabilities: [Apple_MerchantCapabilities]?) -> PKMerchantCapability {
+
+        if(merchantCapabilities == nil){
+            return .capability3DS;
+        }
+
+        var decodedCapabilities: PKMerchantCapability = [];
+
+        if(merchantCapabilities!.contains(Apple_MerchantCapabilities.threeds) ){
```

Comment:
Maybe we should add a guard check for everyone?

Reply:
I added some guard checks on merchantCapabilities being null